### PR TITLE
`azurerm_service_plan` - New SKUs

### LINKED ISSUE #21365: Support for new app service plan SKUs
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

New Azure App Service Plans have been published.

New SKU:s (hopefully all)
P0v3
P1mV3
P2mV3
P3mV3
P4mV3
P5mV3
I4v2
I5v2
I6v2

More info: https://azure.microsoft.com/en-us/blog/new-azure-app-service-plans-fuel-greater-choice-and-savings/

### New or Affected Resource(s)/Data Source(s)

azurerm_service_plan

### Potential Terraform Configuration

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by aristosvo**
Fixes #21365

## AccTests
- [ ] Due to limited availability of the new Service Plans yet (`az appservice list-locations --sku P1mV3`), we'd probably need to make some changes to the test setup

```bash

❯ export ARM_TEST_LOCATION="northeurope"
❯ go install && make acctests SERVICE='appservice' TESTARGS='-run=TestAccServicePlan_memoryOptimized'
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./internal/services/appservice -run=TestAccServicePlan_memoryOptimized -timeout 180m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccServicePlan_memoryOptimized
=== PAUSE TestAccServicePlan_memoryOptimized
=== CONT  TestAccServicePlan_memoryOptimized
--- PASS: TestAccServicePlan_memoryOptimized (122.91s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice124.851s
```

### CODE REVIEW

**Comment by jackofallops** [comment]
Hi @aristosvo - Do you have links to docs supporting the region availability of the new SKU's for context for this pR to be reviewed?

**Comment by aristosvo** [comment]
[Based on the announcement: H1](https://azure.microsoft.com/en-us/blog/new-azure-app-service-plans-fuel-greater-choice-and-savings/).

It is available in ~6 regions at the moment, based on `az appservice list-locations --sku P1mV3`.

File: internal/services/appservice/service_plan_resource_test.go [hunk 1]
```diff
@@ -379,6 +394,36 @@ resource "azurerm_service_plan" "test" {
...
+  worker_count             = 3
+
+  zone_balancing_enabled = true
+
+  tags = {
+    environment = "AccTest"
+    Foo         = "bar"
+  }
+}
+`, data.RandomInteger, data.Locations.Primary)
```
Comment (jackofallops) [inline internal/services/appservice/service_plan_resource_test.go:424]: Could you update this to:
```suggestion
`, data.RandomInteger, data.Locations.Secondary)
```
and I'll PR a change for the CI setup to support the testing of this SKU?

**Review by jackofallops — CHANGES_REQUESTED** [review]
Thanks @aristosvo - one comment below to address and I'll PR the pre-requisite change so we can run the new test too.

Thanks

File: internal/services/appservice/service_plan_resource_test.go (hunk 1 above)
Comment (aristosvo) [reply internal/services/appservice/service_plan_resource_test.go]: Sure!

**Review by jackofallops — APPROVED** [review]
Thanks @aristosvo - LGTM 👍

**Pull request merged**